File: api-service-ms/src/main/java/com/fpoly/ooc/entity/Discount.java
package com.fpoly.ooc.entity;

import com.fpoly.ooc.request.promotion.DiscountRequest;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "discount")
@Entity
public class Discount extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "discount_code")
    private String discountCode;

    @Column(name = "discount_name")
    private String discountName;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "end_date")
    private LocalDateTime endDate;

    @Column(name = "discount_value")
    private BigDecimal discountValue;

    @Column(name = "discount_max_value")
    private BigDecimal discountMaxValue;

    @Column(name = "discount_method")
    private String discountMethod;

    @Column(name = "discount_condition")
    private BigDecimal discountCondition;

    @Column(name = "apply_for")
    private String applyFor;

    @OneToMany(mappedBy = "discountId")
    private List<DiscountProduct> discountProductList;

}

File: fe-ms/src/components/admins/promition/Discount.js
import styles from "./Discount.module.css";

function Discount({ data }) {
  const {
    id,
    discountCode,
    discountName,
    startDate,
    endDate,
    discountValue,
    discountMaxValue,
    discountMethod,
    discountCondition,
    status,
  } = data;

  return (
    <tr className={styles.trow}>
      <th scope="row">1</th>
      <td>{discountName}</td>
      <td>
        {discountValue} {`${discountMethod === "Money" ? "vnd" : "%"}`}
      </td>
      <td>
        {discountMaxValue} {`${discountMethod === "Money" ? "vnd" : "%"}`}
      </td>
      <td>{discountCondition}</td>
      <td>{discountMethod}</td>
      <td>
        {startDate} - {endDate}
      </td>
      <td>{`${status === "ACTIVE" ? "Kích hoạt" : "Hủy kích hoạt"}`}</td>
      <td>edit</td>
    </tr>
  );
}

export default Discount;
